### ISSUE TITLE
Audacity sometimes fails to stop at then end of a selection while overdubbing on Windows

### ISSUE THREAD

**Issue by JohnColket** [audacity/audacity#3898]
### Bug description

Forum user haqueshenas https://forum.audacityteam.org/viewtopic.php?p=458530#p458530 complained that when overdubbing, Audacity does not automatically stop the recording.

### Steps to reproduce

1) Startup Audacity
2) Tools > Reset Configuration
3) Generate 8 second DTMF
4) Transport > Recording > New Track

Note that Audacity fails to stop recording at 8 seconds

### Expected behavior

Audacity stops recording at 8 seconds.

### Actual behavior

Audacity continues to record and the recording must be stopped manually.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

This behavior is present in 3.2.1 and seems to have started in 3.1.0, although it is difficult to tell as the behavior is inconsistent.

Steve says it works properly for him on Linux.  OP says reducing the clip size to less than 3 seconds fixes the problem.  When I tested, it sometimes failed and sometimes worked at 1 second.  Similarly, it sometimes failed and sometimes worked at 8 seconds.

**Comment by DavidBailes** [comment]
I think this has been fixed by https://github.com/audacity/audacity/commit/7a2df4fe9ad26bf02162f89ecdb4a7317c854ff8, which has been merged into 3.2.2.

**Comment by petersampsonaudacity** [comment]
Testing on W10 with 3.2.1 this **_only_** fails for me if the DTMF tone is not selected after step 3.

If the DTMF tone track is **_fully selected_** after Step 3 the the new track recording properly stops recording after 8 seconds

**Comment by petersampsonaudacity** [comment]
The original poster complained on the Forum that it failed if the selection was less than one second - so I tested that on W10 with 3.2.1 and that worked fine:

![image](https://user-images.githubusercontent.com/50205984/200529947-f30bccb2-dfda-4a73-92a4-222ce28bb656.png)

**Comment by petersampsonaudacity** [comment]
I also just tested  on 3.2.1 with a 3 minute selection and the recording properly stopped at 3 minutes.

But what I am finding is that sometimes (but not always, it's moonphase) although the recording stops Audacity is left with the Record button down (is this an issue that you fixed for 3.2.2?).

**Comment by Ryan-Miller-Audacity** [comment]
Testing this in the latest 3.2.2 build (https://github.com/audacity/audacity/actions/runs/3603954269) as Peter mentions, I am seeing this work correctly. If the track is selected (either fully or a segment), the recording properly stops at the end of the segment and the recording visual state is stopped. If the track is not selected recording continues on forever (as one would expect).
